.circleci/config.yml

```diff
@@ -0,0 +1,95 @@
+version: 2
+
+# this circle ci config is Inspired by minishift's scripts
+# https://github.com/minishift/minishift/blob/cfc599cffdc41d31b207f7cfd40190be6b86b858/.circleci/config.yml
+# and an orb defined at https://github.com/snowdrop/circleci-orbs for provisioning the oc cluster
+
+jobs:
+  build:
+    machine: true
+    environment:
+      GOVERSION: 1.11.4
+    working_directory: ~/go/src/github.com/redhat-developer/odo
+
+    steps:
+    - checkout
+    - run:
+        name: Settingup machine with golang
+        command: |
+            echo 'export GOPATH=$HOME/go' >> $BASH_ENV
+            echo 'export PATH=$GOPATH/bin:/usr/local/go/bin/:$PATH' >> $BASH_ENV
+            sudo rm -rf /usr/local/go
+            curl https://storage.googleapis.com/golang/go${GOVERSION}.linux-amd64.tar.gz | sudo tar -C /usr/local -xz
+            go env
+            mkdir -p $GOPATH/bin
+
+    - run: 
+        name: Build Binary
+        command: |
+            go version
+            env
+            cd $GOPATH/src/github.com/redhat-developer/odo
+            make goget-tools
+            make
+            sudo cp odo /usr/bin
+    
+    - run:
+        name: "Download, install okd, kubectl tools, launch okd and grant cluster admin role for the admin user"
+        command: |
```

Comment:
Why we can't use the same script as we have for travis? 
./scripts/oc-cluster.sh

Reply:
I consumed an orb which georgies shared(the one they are using here[1] ) during the initial runs in my fork and it successfully ran all tests which do not require `service-catalog`.  
so pushed it here for testing, now started a test using `oc-cluster.sh` script and added all the test suite.

[1] https://github.com/snowdrop/spring-boot-bom/blob/sb-1.5.x/.circleci/config.yml#L7
[2] https://circleci.com/orbs/registry/orb/snowdrop/openshift#commands-launch_openshift_311

Reply:
it seems like there are errors[1] while trying with `scripts/oc-cluster.sh` 
so for now we can run test suites which doesnot require service-catalog on circlieCI.

[1] https://circleci.com/gh/syamgk/odo/55?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

Reply:
I'm confused, why not reuse the orb? If you don't, then it probably makes sense to just use the script within odo to launch the cluster

Reply:
Just saw that you are not using the orb because it requires enabling them. IMHO that shouldn't be a problem :)

Reply:
I'm confused as well…

Reply:
@metacosm @geoand , here this its using the same scripts which is on that orb openshift_311.
since the other script `oc-cluster.sh` is erroring as mentionend here https://github.com/redhat-developer/odo/pull/1308/files#r255052209

Reply:
OK, but why not use the orb itself? It's just a matter of configuring the org and/or repo settings right?